### ISSUE TITLE
[Bug] Loss of items

### ISSUE THREAD

**Issue by p75369** [gr8pefish/IronBackpacks#19]
I had items destroyed when using Quick Deposit to dump my backpack into a Storage Drawers controller that did not enough attached drawers to store everything.

Minecraft version 1.7.10
Iron Backpacks version 1.0.4
Forge version/build 7.10.99.99 (i think)
Versions of any mods potentially related to the issue Storage Drawers 1.5.14

Steps to reproduce
Place down an insufficient number of storage drawers with an attached controller (mainly focussing on the number of different types of item, rather than stack size), shift click the contents of the backpack into the controller.

**Comment by gr8pefish** [comment]
Thanks for the detailed bug report.

I will look into the issue. I am currently really busy with school and exams, but in 3 weeks or so I should finally have free time again (I hope xD). So for now maybe just be careful with that upgrade (it seems I didn't do enough checks on it).

Thanks!

**Comment by gr8pefish** [comment]
I tried to replicate this bug (using the latest version of Storage Drawers - 1.6.2) and was unable to do so. Could you try updating your version of Storage Drawers and trying it again to see if it still happens? And if it does still happen please be more detailed in what exactly you did. 

Thanks, I want to get this fixed :)

**Comment by Hackmysack** [comment]
Just updated to Storage Drawers 1.6.2 with Iron Backpacks 1.0.4 and was still able to replicate:
1: Put stuff in backpack
2: Shift right click on storage controller
3: Check backpack
4: Be amazed at magic trick/empty backpack.

Also to note I had a few drawers free, but i think its just the controller

Update: Did another test with a fresh set of draws _forgot to test with locked draws_ but had 4 singles and one 4 draw block and a controller moved them all outside.... and it worked so maybe it's only with a large array of draws or something else im missing.

**Comment by gr8pefish** [comment]
Yes, I am still having difficulty, doing steps 1-4 doesn't void anything for me. Can you tell me **exactly** how to replicate an example of where items disappear (including number of drawers, if they are locked, etc.)?

Are you on a single player or on a dedicated server?
Are there any other mods that may be conflicting? Modpack?

**Comment by gr8pefish** [comment]
By the way @TehNut, any ideas why this (may have) happened?

**Comment by TehNut** [comment]
The only thing I can think of is that the Drawer Controller reports X open slots when you right click it, the backpack thinks it has enough so it dumps all the items, then the controller actually doesn't have the amount that it reported, so the items poof. 

tldr: I'm fairly sure this is on StorageDrawers' side, If it works fine with other inventories that don't have enough items.

**Comment by gr8pefish** [comment]
Yeah that's what I was thinking as well, I double checked my code and I thought it was fine, but it's always good to have a second opinion. I just tried breaking it by using the quick deposit upgrade on a different small inventory (chest) and there weren't any issues, so it looks like you are right. I'll make sure jaquerdo knows about it.

To the others: I'm closing the issue, but if you can easily replicate it again let me know and I will reopen it.
